shell/components/Resource/Detail/PercentageBar.vue

```diff
@@ -0,0 +1,45 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+export interface Props {
+  percent: number;
+  percentageColor?: string;
+  baseColor?: string;
+}
+
+const props = withDefaults(
+  defineProps<Props>(),
+  {
+    baseColor:       'var(--progress-bg)',
+    percentageColor: 'var(--primary)'
+  }
+);
+
+const baseStyle = computed(() => ({ backgroundColor: props.baseColor }));
+const usedStyle = computed(() => ({
+  backgroundColor: props.percentageColor,
+  width:           `${ props.percent }%`
+}));
+</script>
+
+<template>
+  <div
+    class="percentage-bar"
+    :style="baseStyle"
```

Comment:
optional: we can consider using `v-bind` in CSS instead of binding to inline styles[^1].

[^1]: https://vuejs.org/api/sfc-css-features.html#v-bind-in-css

Reply:
I didn't know about that feature. Thanks, I've updated it. I know there's some other style props in this pr but I don't think it works in the other locations.